src/data/zmod/basic.lean

```diff
@@ -577,6 +577,79 @@ def units_equiv_coprime {n : ℕ} [fact (0 < n)] :
   left_inv := λ ⟨_, _, _, _⟩, units.ext (nat_cast_zmod_val _),
   right_inv := λ ⟨_, _⟩, by simp }
 
+/-- A ring `R` isomorphic to `R × zmod 1` -/
+def prod_zmod_one_equiv (R : Type*) [semiring R] : R ≃+* R × zmod 1 :=
+{ to_fun := λ x, (x, 0),
+  inv_fun := prod.fst,
+  map_add' := by intros; ext; simp; congr,
+  map_mul' := by intros; ext; simp; congr,
+  left_inv := λ x, rfl,
+  right_inv := λ x, by apply prod.ext; simp; congr }
+
+@[simp] lemma prod_zmod_one_equiv_apply {R : Type*} [semiring R] (x : R) :
```

Comment:
these will be made by `@[simps]` on the def

Reply:
Done